File: src/main/java/leets/leenk/domain/leenk/domain/service/LeenkParticipantsGetService.java
package leets.leenk.domain.leenk.domain.service;

import java.util.List;
import leets.leenk.domain.leenk.application.exception.LeenkParticipantNotFoundException;
import leets.leenk.domain.leenk.domain.entity.Leenk;
import leets.leenk.domain.leenk.domain.entity.LeenkParticipants;
import leets.leenk.domain.leenk.domain.repository.LeenkParticipantsRepository;
import leets.leenk.domain.user.domain.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LeenkParticipantsGetService {

    private final LeenkParticipantsRepository leenkParticipantsRepository;

    public List<LeenkParticipants> findAllByLeenk(Leenk leenk) {

        return leenkParticipantsRepository.findAllByLeenk(leenk);
    }

    public Slice<LeenkParticipants> findSliceByParticipant(User user, Pageable pageable) {
        return leenkParticipantsRepository.findAllByParticipantOrderByJoinedAtDesc(user, pageable);
    }

    public boolean existsByLeenkAndParticipant(Leenk leenk, User user) {
        return leenkParticipantsRepository.existsByLeenkAndParticipant(leenk, user);
    }

    public LeenkParticipants findByLeenkAndParticipantId(Long leenkId, Long participantId) {

        return leenkParticipantsRepository.findByLeenkIdAndParticipantId(leenkId, participantId)
                .orElseThrow(LeenkParticipantNotFoundException::new);
    }
}

File: src/main/kotlin/leets/leenk/domain/leenk/domain/service/LeenkParticipantsGetService.kt
package leets.leenk.domain.leenk.domain.service

import leets.leenk.domain.leenk.application.exception.LeenkParticipantNotFoundException
import leets.leenk.domain.leenk.domain.entity.Leenk
import leets.leenk.domain.leenk.domain.entity.LeenkParticipants
import leets.leenk.domain.leenk.domain.repository.LeenkParticipantsRepository
import leets.leenk.domain.user.domain.entity.User
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import org.springframework.stereotype.Service

@Service
class LeenkParticipantsGetService(
    private val leenkParticipantsRepository: LeenkParticipantsRepository,
) {
    fun findAllByLeenk(leenk: Leenk): List<LeenkParticipants> =
        leenkParticipantsRepository.findAllByLeenk(leenk)

    fun findSliceByParticipant(user: User, pageable: Pageable): Slice<LeenkParticipants> =
        leenkParticipantsRepository.findAllByParticipantOrderByJoinedAtDesc(user, pageable)

    fun existsByLeenkAndParticipant(leenk: Leenk, user: User): Boolean =
        leenkParticipantsRepository.existsByLeenkAndParticipant(leenk, user)

    fun findByLeenkAndParticipantId(leenkId: Long, participantId: Long): LeenkParticipants =
        leenkParticipantsRepository.findByLeenkIdAndParticipantId(leenkId, participantId)
            .orElseThrow { LeenkParticipantNotFoundException() }
}
